src/fsharp/FSharp.Core/printf.fs

```diff
@@ -1267,18 +1267,53 @@ module internal PrintfImpl =
                 let v = get(key.Value)
                 Cache<'T, 'State, 'Residue, 'Result>.last <- (key.Value, v)
                 v
+    
+    let [<Literal>] MAX_BUILDER_SIZE = 360
+    let [<Literal>] MIN_BUILDER_CAPACITY = 128
+    type StringBuilderCache() =
+        // The value 360 was chosen in discussion with performance experts as a compromise between using
+        // as litle memory (per thread) as possible and still covering a large part of short-lived
+        // StringBuilder creations on the startup path of VS designers.
+       
+        [<ThreadStatic; DefaultValue>]
+        static val mutable private CachedInstance: StringBuilder
+ 
+        static member Acquire(capacity) =
+            if capacity <= MAX_BUILDER_SIZE then
+                let sb = StringBuilderCache.CachedInstance
+                if not (isNull sb) && capacity <= sb.Capacity then
+                    // Avoid stringbuilder block fragmentation by getting a new StringBuilder
+                    // when the requested size is larger than the current capacity
+                    StringBuilderCache.CachedInstance <- null
+                    sb.Clear() |> ignore
+                    sb
+                else
+                    let capacity = max capacity MIN_BUILDER_CAPACITY
+                    new StringBuilder(capacity)
+            else
+                let capacity = max capacity MIN_BUILDER_CAPACITY
+                new StringBuilder(capacity)
+ 
+        static member Release(sb:StringBuilder) =
+            if sb.Capacity <= MAX_BUILDER_SIZE then
```

Comment:
Check if `sb` is not `null`?

Reply:
Good idea. The original code should do that as well ;-)